fit: optimize.Local removed from gonum

See https://github.com/gonum/gonum/pull/538

Getting `go/src/go-hep.org/x/hep/fit/curve1d.go:28:9: undefined: optimize.Local` when running `go get -insecure go-hep.org/x/hep/fit` (in the gophernotes `Dockerfile.DS`).

--- a/fit/curve1d.go
+++ b/fit/curve1d.go
@@ -10,7 +10,7 @@ import (
 
 // Curve1D returns the result of a non-linear least squares to fit
 // a function f to the underlying data with method m.
-func Curve1D(f Func1D, settings *optimize.Settings, m optimize.Method) (*optimize.Result, error) {
+func Curve1D(f Func1D, settings *optimize.Settings, m optimize.GlobalMethod) (*optimize.Result, error) {
 	f.init()
 
 	p := optimize.Problem{
@@ -25,5 +25,9 @@ func Curve1D(f Func1D, settings *optimize.Settings, m optimize.Method) (*optimiz
 
 	p0 := make([]float64, len(f.Ps))
 	copy(p0, f.Ps)
-	return optimize.Local(p, p0, settings, m)
+	if settings == nil {
+		settings = optimize.DefaultSettings()
+	}
+	settings.InitX = p0
+	return optimize.Global(p, len(p0), settings, m)
 }

--- a/fit/hist.go
+++ b/fit/hist.go
@@ -14,7 +14,7 @@ import (
 // Only bins with at least an entry are considered for the fit.
 // In case settings is nil, the optimize.DefaultSettings is used.
 // In case m is nil, the same default optimization method than for Curve1D is used.
-func H1D(h *hbook.H1D, f Func1D, settings *optimize.Settings, m optimize.Method) (*optimize.Result, error) {
+func H1D(h *hbook.H1D, f Func1D, settings *optimize.Settings, m optimize.GlobalMethod) (*optimize.Result, error) {
 	var (
 		n     = h.Len()
 		xdata = make([]float64, 0, n)

